spec/support.rb

```diff
@@ -1,3 +1,7 @@
 def remove_file(file_path)
   File.delete(file_path) if File.exist?(file_path)
 end
+
+def version_valid?(version)
```

Comment:
We can leverage RSpec custom matchers here to clean up the tests a little.  Helper functions in specs are often a fair signal that you might have an opportunity for a matcher.
This
`expect(version_valid?(scanner.version)).to be_truthy
`
could be
`expect(scanner.version).to be_a_valid_version
`
Given
```
RSpec::Matchers.define :be_a_valid_version do
  match do |actual|
     /^\d+\.\d+(.\d+)*$/ =~ actual
  end
end
```

Reply:
very good suggestion. Modified accordingly.